### PR TITLE
add MaskAucCalculator in paddlebox

### PR DESCRIPTION
**Pull request by danleifeng**
add MaskAucCalculator in paddlebox.
It can calculate the partial AUC where mask_value = 1.

### CODE REVIEW

File: paddle/fluid/framework/fleet/box_wrapper.h [hunk 1]
```diff
@@ -413,6 +413,42 @@ class BoxWrapper {
...
+      get_data<int64_t>(exe_scope, label_varname_, &label_data);
+      std::vector<float> pred_data;
+      get_data<float>(exe_scope, pred_varname_, &pred_data);
+      std::vector<int64_t> mask_data;
+      get_data<int64_t>(exe_scope, mask_varname_, &mask_data);
+      auto cal = GetCalculator();
+      auto batch_size = label_data.size();
+      std::string out = "";
+      for (size_t i = 0; i < batch_size; ++i) {
+        out += std::to_string(mask_data[i]) + " ";
```
Comment (hutuxian) [inline paddle/fluid/framework/fleet/box_wrapper.h]: what dose variable 'out' mean?

Comment (danleifeng) [reply paddle/fluid/framework/fleet/box_wrapper.h]: used for testing. Deleted it done. Thanks!

File: paddle/fluid/framework/fleet/box_wrapper.h [hunk 2]
```diff
@@ -425,7 +457,7 @@ class BoxWrapper {
                   const std::string& pred_varname,
                   const std::string& cmatch_rank_varname, bool is_join,
                   const std::string& cmatch_rank_group,
-                  int bucket_size = 1000000) {
+                  const std::string& mask_varname, int bucket_size = 1000000) {
```
Comment (gavin1332) [inline paddle/fluid/framework/fleet/box_wrapper.h]: Incompatible modification. Have you ever tested other AUC calculator working fine?

**Review by gavin1332 — COMMENTED** [review]
Incompatible to current calculator initialization

File: paddle/fluid/framework/fleet/box_wrapper.h (hunk 2 above)
Comment (danleifeng) [reply paddle/fluid/framework/fleet/box_wrapper.h]: yes. re-ordered the params. Thanks!

**Review by gavin1332 — APPROVED** [review]
LGTM

**Pull request merged**